src/core/ItemAtt.cpp

```diff
@@ -603,3 +603,118 @@ bool CItemAtt::Matches(time_t time1, time_t time2, int iObject,
     return PWSMatch::Match(time1, time2, testtime, iFunction);
   }
 }
+
+void CItemAtt::SerializePlainText(vector<char> &v)  const
+{
+  uuid_array_t uuid_array;
+  time_t t = 0;
+    
+  v.clear();
+  
+  // write mandatoty field ATTUUID
+  v.push_back(ATTUUID);
+  GetUUID(uuid_array);
+  CItem::push_length(v, sizeof(uuid_array_t));
+  v.insert(v.end(), uuid_array, (uuid_array + sizeof(uuid_array_t)));
+    
+  if(IsTitleSet())
+    push(v, ATTTITLE, GetTitle());
+    
+  if(IsCreationTimeSet()) {
+    GetCTime(t);
+    push(v, ATTCTIME, t);
+  }
+
+  if(IsFieldSet(MEDIATYPE))
+    push(v, MEDIATYPE, GetMediaType());
+  if(IsFieldSet(FILENAME))
+    push(v, FILENAME, GetFileName());
+  if(IsFieldSet(FILEPATH))
+    push(v, FILEPATH, GetFilePath());
+    
+  if(HasContent()) {
+    size_t length = GetContentSize();
+    size_t realLen = GetContentLength();
+    if(length) {
+      unsigned char data[length];
+      if(GetContent(data, length)) {
+        v.push_back(CONTENT);
+        push_length(v, static_cast<uint32>(realLen));
+        v.insert(v.end(), data, (data + realLen));
+      }
+    }
+  }
+    
+  if(IsFieldSet(FILECTIME)) {
+    GetFileCTime(t);
+    push(v, FILECTIME, t);
+  }
+  if(IsFieldSet(FILEMTIME)) {
+    GetFileMTime(t);
+    push(v, FILEMTIME, t);
+  }
+  if(IsFieldSet(FILEATIME)) {
+    GetFileATime(t);
+    push(v, FILEATIME, t);
+  }
+    
+  int end = END; // just to keep the compiler happy...
+  v.push_back(static_cast<char>(end));
+  push_length(v, 0);
+}
+
+
+bool CItemAtt::DeSerializePlainText(const std::vector<char> &v)
```

Comment:
Has this function been tested for memory corruption and other potential security weaknesses? Even if yes, without unit tests how can we ensure no bugs will be introduced in future development?

Reply:
This type of code is similar to the „old“ code in ItemData.cpp. It is tested several times and no problem occurred in Modultest.